site/src/pages/release-notes/1.25.0.mdx

```diff
@@ -0,0 +1,153 @@
+    .decorator(ObservationService.newDecorator(observationRegistry))
+  ...
+  ```
+- **WebSocket Client Support**: You can now send and receive data over [WebSocket](https://en.wikipedia.org/wiki/WebSocket)
+  using <type://WebSocketClient>. #4972
+  ```java
+  WebSocketClient client = WebSocketClient.of("ws://...");
+  client.connect("/").thenAccept(webSocketSession -> {
+    WebSocketWriter writer = WebSocket.streaming();
+    webSocketSessions.setOutbound(writer);
+    outbound.write("Hello!");
+
+    Subscriber<WebSocketFrame> subscriber = new Subscriber<WebSocketFrame>() {
+      ...
+    }
+    webSocketSessions.inbound().subscribe(subscriber);
+  });
+  ```
+- **Implement gRPC Richer Error Model more easily**: You can now easily use gRPC
+  [Richer Error Model](https://grpc.io/docs/guides/error/#richer-error-model) via <type://GoogleGrpcStatusFunction>. #4614 #4986
+  ```java
+  GoogleGrpcStatusFunction statusFunction = (ctx, throwable, metadata) -> {
+    if (throwable instanceof MyException) {
+      return com.google.rpc.Status.newBuilder()
+        .setCode(Code.UNAUTHENTICATED.getNumber())
+        .addDetails(detail(throwable))
+        .build();
+    }
+    ...
+  };
+  Server.builder().service(
+    GrpcService.builder()
+      .exceptionMapping(statusFunction))
+  ```
+- **Set HTTP Trailers Easily** You can now easily set trailers to be sent after the data stream using
+  <type://HttpRequest#of(RequestHeaders,Publisher,HttpHeaders)> or
+  <type://HttpResponse#of(ResponseHeaders,Publisher,HttpHeaders)>. #3959 #4727
+- **New API for Multipart Headers**: You can now retrieve headers from a multipart request in an annotated service
+  using <type://MultipartFile#headers()>. #5106
+- **Access RequestLogProperty values more easily**: <type://RequestLogAccess#getIfAvailable()>
+  has been introduced, which allows users to access a <type://RequestLogProperty> immediately if available. #4956 #4966
+- **Keep an Idle Connection Alive on PING**: The `keepAliveOnPing` option has been introduced. Enabling this option will keep
+  an idle connection alive when an HTTP/2 PING frame or `OPTIONS * HTTP/1.1` is received. The option can be configured
+  by <type://ServerBuilder#idleTimeout(Duration,boolean)> or <type://ClientFactoryBuilder#idleTimeout(Duration,boolean)>. #4794 #4806
+- **Create a StreamMessage from Future**: You can now easily create a <type://StreamMessage> from a `CompletionStage`
+  using <type://StreamMessage#of(CompletionStage<?>)>. #4995
+
+## 📈 Improvements
+
+- The number of event loops is equal to the number of cores by default when `io_uring` is used as the transport type. #5089
+- You can now customize error responses when a service for a request is not found
+  using <type://ServiceErrorHandler#renderStatus()>. #4996
+- Redirection for a trailing slash is done correctly even if a reverse proxy rewrites the path. #4994
+- <type://DocService> now tries to guess the correct route behind a reverse proxy. #4987
+
```

Comment:
```suggestion
- The `RetentionPolicy` of <type://@UnstableApi> annotation is now `CLASS` so that
  bytecode analysis tools can detect the declaration and usage of unstable APIs. #5131

```